### PR TITLE
Sorts quirks into medical record categories

### PR DESCRIPTION
**Pull request by RaveRadbury**
## About The Pull Request
Back in  #44009 I dumped quirks into the notes of the med record. Now I'm sorting Quirks with a value of -2 or higher into Major Disabilities and Quirks with a value of -1 into Minor Disabilities (I might also add neutral quirks to minor disabilities) Everything else goes into notes.

I removed allergens because we have no mechanical way to enforce them, if anyone has strong reasons to keep it I'd be interested in hearing them. Apparently the disease category is supposed to populate via added comments, but I have no idea.

## Why It's Good For The Game
This further improves the medical record system.

## Changelog
:cl:
tweak: Quirks sort into medical records rather than just dumping into notes.
/:cl:

Currently there is a bug where if a disability category is filled in, the ones that come after it are ignored, so if there are Major Disabilities only those categories are populated and Minor Disabilities and Notes aren't. If there are no Major Disabilities are present, then Minor Disabilities are filled. I need help figuring this out.

### CODE REVIEW

**Comment by Mickyan** [comment]
Ah, actually seeing this in context now I can't really find what the issue could be

I think maybe you want to turn those categories into defines though

File: code/datums/traits/_quirk.dm [hunk 1]
```diff
@@ -65,22 +65,78 @@
...
-		return dat.Join(", ")
-	else
-		for(var/V in roundstart_quirks)
-			var/datum/quirk/T = V
-			dat += T.medical_record_text
-		if(!dat.len)
-			return "None"
-		return dat.Join("<br>")
+	switch(category)
+		if(0)
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/_quirk.dm:71]: magic numbers

File: code/datums/traits/_quirk.dm [hunk 2]
```diff
@@ -65,22 +65,78 @@
...
+					if(T.value > -1)
+						dat += T.name
+					if(!dat.len)
+						return "None"
+				return dat.Join(", ")
+			else
+				for(var/V in roundstart_quirks)
+					var/datum/quirk/T = V
+					if(T.value > -1)
+						dat += T.medical_record_text
```
Comment (Rohesie) [inline code/datums/traits/_quirk.dm:136]: Your way is better performant, but I'd use ternary operators for higher clarity and less duplicate code.
Examples:
`dat += medical ? T.medical_record_text : T.name`
`return medical ? "No minor disabilities have been declared." : "None"`

Comment (RaveRadbury) [reply code/datums/traits/_quirk.dm]: Thank you!

File: code/datums/traits/_quirk.dm (hunk 1 above)
Comment (Rohesie) [reply code/datums/traits/_quirk.dm:71]: Essentially to fix this you need to do defines for the categories.
Something like
```
#define CAT_MINOR_DISSABLIITY 2
#define CAT_MAJOR_DISSABLIITY 3
```
Then you replace the `if(2)` for `if(CAT_MINOR_DISSABLIITY)` and it's more readable.
The exact name of the define is up to you, just follow the convention of making it all in caps, try for them to have the same beginning or end, and try to make it unique enough not to risk being duplicated in the future.

**Pull request merged**